plugins/module_utils/network/sonic/config/bgp_neighbors/bgp_neighbors.py

```diff
@@ -420,13 +428,44 @@ def build_bgp_peer_groups_payload(self, cmd, have, bgp_as, vrf_name):
                     if peer_group['address_family'].get('afis', None) is not None:
                         for each in peer_group['address_family']['afis']:
                             samp = {}
+                            afi_safi_cfg = {}
+                            pfx_lmt_cfg = {}
+                            pfx_lst_cfg = {}
+                            ip_dict = {}
                             if each.get('afi', None) is not None and each.get('safi', None) is not None:
                                 afi_safi = each['afi'].upper() + "_" + each['safi'].upper()
                                 if afi_safi is not None:
                                     afi_safi_name = 'openconfig-bgp-types:' + afi_safi
                                 if afi_safi_name is not None:
                                     samp.update({'afi-safi-name': afi_safi_name})
                                     samp.update({'config': {'afi-safi-name': afi_safi_name}})
+                                if afi_safi == 'IPV4_UNICAST':
+                                    if each.get('ip_afi', None) is not None:
+                                        afi_safi_cfg = self.get_afi_safi_config_payload(each['ip_afi'])
+                                        if afi_safi_cfg:
+                                            ip_dict.update({'config': afi_safi_cfg})
+                                    if each.get('prefix_limit', None) is not None:
+                                        pfx_lmt_cfg = self.get_prefix_limit_payload(each['prefix_limit'])
+                                        if pfx_lmt_cfg:
+                                            ip_dict.update({'prefix-limit': {'config': pfx_lmt_cfg}})
+                                    if ip_dict:
```

Comment:
Since "ip_dict" has been initialized to {}, (which is not "null"), this will always be True. 

I think that the intent of this instruction would be implemented as:

if ip_dict != {}

This same comment applies to the similar cases (above) for afi_safi_cfg and pfx_lmt_cfg, as well as the case below for pfx_lst_cfg.